src/macrofier/engine.mjs

```diff
@@ -52,7 +52,8 @@ let types = {
   getMethodSignatureParams: ()=>null,
   getSchemaShape: ()=>null,
   getSchemaType: ()=>null,
-  getJsonType: ()=>null
+  getJsonType: ()=>null,
+  getMethodImpl: ()=>null
```

Comment:
I really want to avoid returning an entire method from a function, rather than a template.

I looked at some of the different properties, e.g. array vs string, and I think what we need is a new macro called: `${result.instantiation}` where we can include things like:

```c
if (resultNam != nullPtr) {
  FireboltSDK::JSON::String* strResult = new FireboltSDK::JSON::String(jsonResult);
   *deviceType = static_cast<FireboltTypes_StringHandle>(strResult);  
}
```

vs


```c
        if (videoResolution != nullptr) {
            WPEFramework::Core::ProxyType<WPEFramework::Core::JSON::ArrayType<WPEFramework::Core::JSON::DecSInt32>>* resultPtr = new WPEFramework::Core::ProxyType<WPEFramework::Core::JSON::ArrayType<WPEFramework::Core::JSON::DecSInt32>>();
            *resultPtr = WPEFramework::Core::ProxyType<WPEFramework::Core::JSON::ArrayType<WPEFramework::Core::JSON::DecSInt32>>::Create();
            *(*resultPtr) = jsonResult;
            *videoResolution = static_cast<Device_ResolutionIntegerArrayHandle>(resultPtr);
        }
```


Then we can create a method called `getSchemaInstantiation()` and use it to populate that macro from a template.